watertap/unit_models/ion_exchange_0D.py

```diff
@@ -0,0 +1,1352 @@
+            units=pyunits.dimensionless,
+            bounds=(0, 1),
+            doc="Pump efficiency",
+        )
+
+        # Fix variables that should be fixed
+        self.resin_diam.fix()
+        self.resin_bulk_dens.fix()
+        self.bed_porosity.fix()
+        self.dimensionless_time.fix()
+        self.lh.fix()
+        self.regen_dose.fix()
+        self.regen_ww.fix()
+        self.regen_sg.fix()
+        self.t_bw.fix()
+        self.bw_rate.fix()
+        self.rinse_bv.fix()
+        self.pump_efficiency.fix()
+
+        tmp_dict = dict(**self.config.property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["parameters"] = self.config.property_package
+        tmp_dict["defined_state"] = True  # inlet block is an inlet
+
+        self.properties_in = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of inlet",
+            default=tmp_dict,
+        )
+
+        # Add outlet and waste block
+        tmp_dict["defined_state"] = False  # outlet and waste block is not an inlet
+        self.properties_out = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of outlet",
+            default=tmp_dict,
+        )
+
+        self.properties_waste = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of waste",
+            default=tmp_dict,
+        )
+
+        # Add ports - oftentimes users interact with these rather than the state blocks
+        self.add_port(name="inlet", block=self.properties_in)
+        self.add_port(name="outlet", block=self.properties_out)
+        self.add_port(name="waste", block=self.properties_waste)
+
+        # Add constraints
+        # =========== EQUILIBRIUM ===========
+
+        @self.Constraint(
+            ion_set,
+            doc="Separation factor calc",
+        )
+        def eq_sep_factor(b, j):
+            return b.R_eq[j] == 1 / b.K_eq[j]
+
+        @self.Constraint(
+            ion_set,
+            doc="Langmuir isotherm",
+        )
+        def eq_langmuir(b, j):
+            if j == target_ion:
+                return b.R_eq[j] == (
+                    b.c_norm[j] * (1 - b.resin_eq_capacity / b.resin_max_capacity)
+                ) / (b.resin_eq_capacity / b.resin_max_capacity * (1 - b.c_norm[j]))
+            else:
+                return Constraint.Skip
+
+        @self.Constraint(doc="Resin equilibrium concentration bounds")
+        def eq_resin_eq_capacity(b):
+            return b.resin_eq_capacity <= b.resin_max_capacity
+
+        @self.Constraint(doc="Bed velocity ub")
+        def eq_vel_bed_ub(b):
+            return b.vel_bed <= b.vel_bed_ratio * b.vel_min
+
+        @self.Constraint(doc="Bed velocity lb")
+        def eq_vel_bed_lb(b):
+            return self.vel_bed >= self.vel_min
```

Comment:
Why do you have inequality constraints? Is this necessary?

Reply:
In an earlier version I had a ``Constraint`` for ``vel_min`` and I found these inequalities helped it solve. I will remove these and see how it works then.